src/target/riscv/riscv.c

```diff
@@ -2205,8 +2177,7 @@ int riscv_openocd_poll(struct target *target)
 			struct target *t = list->target;
 			if (!target_was_examined(t))
 				continue;
-			riscv_info_t *r = riscv_info(t);
-			enum riscv_poll_hart out = riscv_poll_hart(t, r->current_hartid);
+			enum riscv_poll_hart out = riscv_poll_hart(t, target->coreid);
```

Comment:
Curious: Why do we use `target->coreid` in some places and `info->index` on others? For simplicity, could we use one and drop the other? Thanks.

Reply:
Although info->index is only in the riscv013, so I can't easily fix that in riscv.c. I'm planning a future cleanup where we stop passing hart id everywhere and instead just pass the target. E.g. in this call there's no need to pass the hart id at all.